### PR TITLE
Add unlink shim to delete files

### PR DESCRIPTION
**Pull request by christianpoveda**
r? @oli-obk 
This closes the whole create, write, read, delete circle for file handling.

### CODE REVIEW

File: tests/run-pass/file_manipulation.rs [hunk 1]
```diff
@@ -22,4 +21,6 @@ fn main() {
     // Reading until EOF should get the whole text.
     file.read_to_end(&mut contents).unwrap();
     assert_eq!(bytes, contents.as_slice());
+    // Removing file should succeed
+    remove_file(path).unwrap();
```
[the remark this answers is not in the document]
Comment (christianpoveda) [reply tests/run-pass/file_manipulation.rs]: Sadly that needs the `__errno_location()` shim first :S

[the remark this answers is not in the document]
Comment (christianpoveda) [reply tests/run-pass/file_manipulation.rs]: Given that we have the shim now, I added the test

File: tests/run-pass/file_manipulation.rs [hunk 2]
```diff
@@ -22,4 +21,10 @@ fn main() {
...
+    // Opening non-existing file should fail
+    assert!(File::open(path).is_err());
```
Comment (RalfJung) [inline tests/run-pass/file_manipulation.rs]: To properly test errno, IMO it would be a good idea to check that this returns the right kind of error.

Comment (christianpoveda) [reply tests/run-pass/file_manipulation.rs]: https://github.com/pulls

**Pull request merged**